src/app.js

```diff
@@ -42,30 +42,24 @@ app.get('/help', (req, res) => {
     })
 })
 
-app.get('/weather', (req, res) => {
-    if (!req.query.address) {
-        return res.send({
-            error: 'No address provided!'
-        })
-    }
-    geocode.geocode(req.query.address, (error, { latitude, longitude, location } = {}) => {
-        if (error) {
-            return res.send({ error })
-        }
-        forecast.forecast(latitude, longitude, (error, { description, temperature, feelslike, icon }) => {
-            if (error) {
-                return res.send({ error })
-            }
-            res.send({
-                description,
-                temperature,
-                feelslike,
-                icon,
-                location,
-                address: req.query.address
+app.get('/weather', async (req, res) => {
+    try {
+        if (!req.query.address && !req.query.lat && !req.query.lon) {
+            return res.send({
+                error: 'No location provided!'
             })
-        })
-    })
+        }
+        if (req.query.address) {
+            const coordinates = await geocode(req.query.address)
+            const weather = await forecast(coordinates)
+            res.status(200).send(weather)
+        } else {
+            const weather = await forecast(req.query.lat, req.query.lon)
```

Comment:
that must not work
also, here code duplication in next 2 lines
```javascript
          const weather = await forecast(coordinates)
          res.status(200).send(weather)
```

Reply:
I changed this part before I saw the comment.
Changed this:
`const weather = await forecast(req.query.lat, req.query.lon)`
To this:
`const weather = await forecast({ latitude: req.query.lat, longitude: req.query.lon })`

Reply:
Regarding code duplication, how can I change it?
When I get the `res.status(200).send(weather)` part out of the if/else it doesn't work, because "weather is not defined".
To fix that I would need declare `weather` before if/else as empty and the assign value to it.
But it looks weird, since I also need to use either var or let for `weather` to actually be able to assign stuff to it.

Reply:
Fixed.